rest_framework_swagger/docgenerator.py

```diff
@@ -139,29 +151,68 @@ def get_operations(self, api_endpoint, introspector):
             # set default response reference
             if self.default_payload_definition:
                 response_messages['default'] = {
-                    "description": "error payload",
                     "schema": {
                         "$ref": "#/definitions/{}".format(self.default_payload_definition_name)
                     }
                 }
 
+            response_code, response_obj = self._get_default_response_object(operation_method, response_type)
+            response_messages[response_code] = response_obj
+
             # overwrite default and add more responses from docstrings
             response_messages.update(doc_parser.get_response_messages())
 
-            response_messages['200'] = {
-                'description': 'Successful operation',
-                'schema': {
-                    '$ref': '#/definitions/' + response_type
-                } if response_type != 'object' else {
-                    'type': response_type
-                }
-            }
+            # Remove blank response objects - allows yaml to remove default responses
+            for code in response_messages:
```

Comment:
Sketchy modification of collection while iterating on it, could explicitly make a copy of keys before iterating